packages/frontend/src/state/controller/hooks.ts

```diff
@@ -360,36 +359,30 @@ const useNormFactor = () => {
   return normFactor
 }
 
-const useIndex = () => {
-  const address = useAtomValue(addressAtom)
-  const web3 = useAtomValue(web3Atom)
-  const networkId = useAtomValue(networkIdAtom)
+export const useIndex = () => {
+  const { getTwapSafe } = useOracle()
   const [index, setIndex] = useAtom(indexAtom)
-  const contract = useAtomValue(controllerContractAtom)
+  const { ethUsdcPool, weth, usdc } = useAtomValue(addressesAtom)
 
-  useEffect(() => {
-    if (!contract) return
-    getIndex(1, contract).then(setIndex)
-  }, [address, networkId, contract, setIndex])
+  useAppEffect(() => {
+    const getTwapEth = () => {
+      if (ethUsdcPool && weth && usdc) {
+        getTwapSafe(ethUsdcPool, weth, usdc, 1).then((quote) => {
+          setIndex(quote)
```

Comment:
It's better to use a new atom of ethPrice eg: `ethPriceAtom`. Index is actually taken from our contract and it serves separate purpose.